docs/source/references/apollo-config.md

```diff
@@ -0,0 +1,201 @@
+
+```js line=3-4
+module.exports = {
+  client: {
+    includes: ['./imports/**/*.js'],
+    excludes: ['**/__tests__/**/*'],
+  },
+};
+```
+
+### `tagName`
+
+*Optional* –– custom tagged template literal.
+
+When using GraphQL with JavaScript or TypeScript projects, it is common to use the `gql` tagged template literal to write out operations. Apollo tools will be looking through your files for the `gql` tag to extract your queries, so if you use a different template literal, you can configure it like so:
+
+```js line=3
+module.exports = {
+  client: {
+    tagName: 'graphql',
+  },
+};
+```
+
+### `addTypename`
+
+*Optional* –– Apollo will by default add the `__typename` field to all operations automatically.
+
+GraphQL clients like Apollo Client often add the `__typename` field to operations automatically when they're sent over the wire. This can come in really handy for things like caching, but it can be turned off by adding `addTypename: false` to the client config:
+
+```js line=3
+module.exports = {
+  client: {
+    addTypename: false,
+  },
+};
+```
+
+### `clientOnlyDirectives`, `clientSchemaDirectives`
+
+*Optional* –– By default, Apollo projects support the following client-side directives:
+
+- `@client` for local state
+- `@rest` for using apollo-link-rest
+- `@connection` for custom pagination with Apollo Client
+- `@type` for dynamic type names with apollo-link-rest
+
+Client side applications can use custom directives on their queries that aren't meant to be sent to the server. Configuration of client side directives beyond the defaults listed above can be set up like so:
+
+```js line=3-4
+module.exports = {
+  client: {
+    clientOnlyDirectives: ['connection', 'type'],
+    clientSchemaDirectives: ['client', 'rest'],
+  },
+};
+```
+
+`clientOnlyDirectives` are directives that should be stripped out of the operation before being sent to the server. An example of this is the `@connection` directive.
+
+`clientSchemaDirectives` are directives that indicate a portion of the operation that is not meant to be sent to the server. These directives are removed as well as the fields they are placed on. An example of this type of of directive is the `@client` directive.
+
+<h2 id="service-config">Server projects</h2>
+
+Server projects are configured through a top level `service` key in the config.
+
+```js line=2
+module.exports = {
+  service: { ... },
+};
+```
+
+Setting up a `service` in your Apollo config will save you from needing to use flags your Apollo CLI commands like `apollo service:push` and `apollo service:check`. You can set up the schema for your service to load in one of two ways:
```

Comment:
Personal opinion, I'd rather us not refer to flags at all in the config reference. Flags should be functional and expected behavior, but we should treat the config as _the_ way and not introduce any possible confusion or other ways to do things.

>Defining a `service` key in your Apollo config will provide the CLI with the information it needs to perform commands like `service:push` and `service:check`.

Reply:
Love it, will do 👍